woocommerce/woocommerce-android
# Issue/4962 Connection dialog disappears after required update

Closes: #4962 


### Description

The PR:
* Makes `readersStatus` the single source of truth of the connection status. It was causing in some cases that we were triggering ExitWithResult twice
* Cancels coroutine which open the update dialog after it's been opened. It was causing the firing of  the `ShowUpdateInProgress` multiple times

### Testing instructions

1. Make an update required via `TerminalWrapper`
2. Try to collect money for an order
3. Notice that after the required update succeeds collection flow keeps going

### Images/gif



https://user-images.githubusercontent.com/4923871/136378150-cf3e21d4-d9c2-4477-b534-62a4bc8b3859.mp4
